src/utils/ValidatorUtils.ts

```diff
@@ -5,7 +5,7 @@ import { BigNumber } from "../utils";
 
 const AddressValidator = define<string>("AddressValidator", (v) => ethersUtils.isAddress(String(v)));
 const HexValidator = define<string>("HexValidator", (v) => ethersUtils.isHexString(String(v)));
-const BigNumberValidator = define<BigNumber>("BigNumberValidator", (v) => BigNumber.isBigNumber(v));
+export const BigNumberValidator = define<BigNumber>("BigNumberValidator", (v) => BigNumber.isBigNumber(v));
 
 const V3DepositSchema = object({
```

Comment:
todo: Refactor to share some definitions with `V3DepositSchema`. It's unfortunately not composable because it defines all possible fields, but it has the advantage of being more granular in its validation. I would propose to refactor such that it is defined as a product of smaller, composable types.